browser/main/lib/dataApi/attachmentManagement.js

```diff
@@ -342,6 +342,58 @@ function getAbsolutePathsOfAttachmentsInContent (markdownContent, storagePath) {
   return result
 }
 
+/**
+ * @description Copies the attachments to the storage folder and returns the mardown content it should be replaced with
+ * @param {String} markDownContent content in which the attachment paths should be found
+ * @param {String} filepath The path of the file with attachments to import
+ * @param {String} storageKey Storage key of the destination storage
+ * @param {String} noteKey Key of the current note. Will be used as subfolder in :storage
+ */
+function importAttachments (markDownContent, filepath, storageKey, noteKey) {
+  return new Promise((resolve, reject) => {
+    const nameRegex = /(!\[.*?]\()(.+?\..+?)(\))/g
+    let attachPath = nameRegex.exec(markDownContent)
+    const promiseArray = []
+    const attachmentPaths = []
+    const groupIndex = 2
+
+    while (attachPath) {
```

Comment:
I think you should improve your code by eliminating repeating code. Maybe something like this will be shorter.
```js
while (attachPath) {
  let attachmentPath = attachPath[groupIndex];
  attachmentPaths.push(attachmentPath);

  attachmentPath = path.isAbsolute(attachmentPath) ? attachmentPath : path.join(path.dirname(filepath), attachmentPath);
  promiseArray.push(this.copyAttachment(attachmentPath, storageKey, noteKey));

  attachPath = nameRegex.exec(markDownContent);
}
```